debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -103,18 +105,21 @@ public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPoi
 		boolean lockwordPrinted = false;
 		UDATA nestedFieldOffset = new UDATA(0);
 		boolean flatObject = false;
-
+		boolean hasHierarchy = (null != nestingHierarchy) && (nestingHierarchy.length > 0);
 		J9ClassPointer nestedClassHierarchy[] = null;
 
-		if ((null != nestingHierarchy) && (nestingHierarchy.length > 0)) {
+		if (hasHierarchy && !(Pattern.matches("\\[([1-9]\\d*|0)\\]", nestingHierarchy[0]) && nestingHierarchy.length == 1)) {
```

Comment:
Yup, so before my changes, I'd just skipped over the [0] case for the valueTypeHelper.findNestedClassHierarchy case and also incremented the stride after this if statement so I didn't need to run the code through this portion. But now with the changes, this is no longer necessary and is also in-fact incorrect because flat objects would not have their strides properly incremented. I'll remove this.